pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -313,6 +333,86 @@ func (n *networkManager) getGatewayDetails(svc *model.Service) []model.NetworkGa
 	return nil
 }
 
+func (n *networkManager) watchGatewayResources(c *Controller, stop <-chan struct{}) {
+	n.gatewayResourceClient = kclient.NewDelayedInformer(c.client, gvr.KubernetesGateway, kubetypes.StandardInformer, kubetypes.Filter{})
+	registerHandlers(c, n.gatewayResourceClient, "Gateways", n.handleGatewayResource, nil)
+	n.gatewayResourceClient.Start(stop)
+}
+
+// handleGateway resource adds a NetworkGateway for each combination of address and auto-passthrough listener
+// discovering duplicates from the generated Service is not a huge concern as we de-duplicate in NetworkGateways
+// which returns a set, although it's not totally efficient.
+func (n *networkManager) handleGatewayResource(_ controllers.Object, obj controllers.Object, event model.Event) error {
+	gw, ok := obj.(*v1beta1.Gateway)
+	if !ok {
+		return nil
+	}
+
+	if nw := gw.GetLabels()[label.TopologyNetwork.Name]; nw == "" {
+		return nil
+	}
+
+	// we don't want to discover gateways with class "istio-remote" from outside cluster's API servers.
+	// this will only pass for the "config cluster" or "local cluster"
+	if !n.discoverRemoteGatewayResources && gw.Spec.GatewayClassName == "istio-remote" {
```

Comment:
I can not understand this well, though it has a comment.

Is it doing some handling with `istio-remote`  for non config cluster?

Reply:
I need to provide a better explanation for `istio-remote`. 

`Gateway` with `istio-remote`: only discover this from the config cluster. This is a way to reference a gateway that lives in a place that we are not watching the API server for. 

`Gateway` with other classes: discover these anywhere.